src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantShardTest.java

```diff
@@ -0,0 +1,40 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+public final class VariantShardTest {
+
+    @DataProvider(name = "variantShards")
+    public Object[][] variantShards(){
+        List<GATKRead> reads = Arrays.asList(makeRead(1, 300, 1), makeRead(100000, 10, 2), makeRead(299999, 2, 3));
+        List<Iterable<VariantShard>> variantShards = Arrays.asList();
+        variantShards.add(Arrays.asList(new VariantShard(0, "1")));
+        variantShards.add(Arrays.asList(new VariantShard(1, "1")));
+        variantShards.add(Arrays.asList(new VariantShard(2, "1"), new VariantShard(3, "1")));
+        return new Object[][]{
+                {reads, variantShards},
+        };
+    }
+
+    public GATKRead makeRead(int start, int length, int i) {
```

Comment:
I would suggest taking DataflowTestData's makeRead and move that in a utility class, then call it here.